src/com/maddyhome/idea/vim/ex/CommandParser.java

```diff
@@ -170,6 +172,14 @@ public int processCommand(@NotNull Editor editor, @NotNull DataContext context,
       return result | RES_EMPTY;
     }
 
+    // If there is a command alias for the entered text, then process the alias and return that
+    // instead of the original command. We want to do this before the history and then log the
+    // resulting command to the history instead of the alias. This is a bit weird, but it
+    // prevents it from doubling up.
+    if (VimPlugin.getCommand().hasAlias(cmd)) {
```

Comment:
Please add a mechanism to avoid infinite recursion!

Reply:
I just made it so that a command alias cannot reference another alias. If you try to execute an alias that references another alias then an error is returned. This might not be the best way, but the other way that I would approach this would be to set a max recurussion threshold and only return an error if that threshold gets reached. If you have a preference for the threshold then I can rewrite it.